File: 백준/Bronze/2745. 진법 변환/진법 변환.py
import sys

N, B = map(str, sys.stdin.readline().split(' '))
N = list(N)
B = int(B)
res = 0

num = len(N)
for i in range(num):
    word = ord(N[num-i-1])
    if (word < 65):        # 숫자인 경우
        res += (word - 48) * (B ** i)
    else:                       # 알파벳인 경우
        res += (word - 55) * (B ** i)
print(res)

File: 백준/Bronze/2745. 진법 변환/진법 변환.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String N = sc.next();
        char[] arr = N.toCharArray();

        int B = sc.nextInt();
        int res = 0;

        int num = N.length();
        for (int i = 0; i < num; i++) {
            char word = arr[num - i - 1];
            int ascii = (int) word;

            if (ascii < 65) {
                res += (ascii - 48) * Math.pow(B, i);
            } else {
                res += (ascii - 55) * Math.pow(B, i);
            }
        }

        System.out.println(res);
        sc.close();
    }
}
